TESTING.md

```diff
@@ -0,0 +1,195 @@
+    w := httptest.NewRecorder()
+    r := httptest.NewRequest("GET", "/test", nil)
+
+    ctx.SetResponse(w)
+    ctx.SetRequest(r)
+
+    result, err := MyController(ctx)
+    assert.NoError(t, err)
+    assert.Equal(t, http.StatusOK, w.Code)
+}
+```
+
+### Best Practices
+
+1. **Use Table-Driven Tests**
+
+```go
+func TestUserController(t *testing.T) {
+    tests := []struct {
+        name    string
+        body    UserRequest
+        want    UserResponse
+        wantErr bool
+    }{
+        {
+            name: "valid user",
+            body: UserRequest{Name: "John", Email: "john@example.com"},
+            want: UserResponse{ID: "123", Name: "John"},
+        },
+        {
+            name:    "invalid email",
+            body:    UserRequest{Name: "John", Email: "invalid"},
+            wantErr: true,
+        },
+    }
+
+    for _, tt := range tests {
+        t.Run(tt.name, func(t *testing.T) {
+            ctx := fuego.NewMockContext[UserRequest]()
+            ctx.SetBody(tt.body)
+
+            got, err := CreateUser(ctx)
+            if tt.wantErr {
+                assert.Error(t, err)
+                return
+            }
+            assert.NoError(t, err)
+            assert.Equal(t, tt.want, got)
+        })
+    }
+}
+```
+
+2. **Test Error Cases**
+
+```go
+func TestErrorHandling(t *testing.T) {
+    ctx := fuego.NewMockContext[UserRequest]()
+    ctx.SetBody(UserRequest{}) // Empty body should trigger validation error
+
+    _, err := CreateUser(ctx)
+    assert.Error(t, err)
+    assert.Contains(t, err.Error(), "validation failed")
+}
+```
+
+3. **Test Validation Rules**
+
+```go
+func TestValidation(t *testing.T) {
+    ctx := fuego.NewMockContext[UserRequest]()
+    ctx.SetBody(UserRequest{
+        Name:  "", // Required field
+        Email: "invalid-email", // Invalid format
+    })
+
+    _, err := CreateUser(ctx)
+    assert.Error(t, err)
+}
+```
+
+4. **Test Middleware Integration**
+
+```go
+func TestWithMiddleware(t *testing.T) {
+    ctx := fuego.NewMockContext[EmptyBody]()
+    ctx.SetHeader("Authorization", "Bearer valid-token")
+
+    // Test that middleware allows the request
+    result, err := AuthMiddleware(MyProtectedController)(ctx)
+    assert.NoError(t, err)
+
+    // Test that middleware blocks unauthorized requests
+    ctx.SetHeader("Authorization", "invalid-token")
+    _, err = AuthMiddleware(MyProtectedController)(ctx)
+    assert.Error(t, err)
+}
+```
+
+### Tips for Effective Testing
+
+1. Keep tests focused on business logic
+2. Use meaningful test names that describe the scenario
+3. Test both success and failure cases
+4. Use helper functions for common test setup
+5. Test validation rules thoroughly
+6. Mock external dependencies when needed
+7. Use subtests for better organization
+8. Test edge cases and boundary conditions
+
+## Contributing
+
+If you find any issues or have suggestions for improving the testing utilities, please open an issue or submit a pull request.
```

Comment:
Useless to add this information in this page, why did you add it here? It makes no sense!